src/jit/hwintrinsiccodegenxarch.cpp

```diff
@@ -1019,6 +1019,28 @@ void CodeGen::genSSE2Intrinsic(GenTreeHWIntrinsic* node)
             break;
         }
 
+        case NI_SSE2_SetScalarVector128:
+        {
+            assert(baseType == TYP_DOUBLE || baseType == TYP_LONG || baseType == TYP_ULONG);
+            assert(op2 == nullptr);
+
+            instruction ins = Compiler::insOfHWIntrinsic(intrinsicID, node->gtSIMDBaseType);
+            if (op1Reg == targetReg)
+            {
+                regNumber tmpReg = node->GetSingleTempReg();
+
+                // Ensure we aren't overwriting targetReg
+                assert(tmpReg != targetReg);
+
+                emit->emitIns_R_R(INS_movapd, emitTypeSize(TYP_SIMD16), tmpReg, op1Reg);
```

Comment:
Use `ps` instructions when they are equivalent to `pd` instructions. They have shorted encoding when VEX is not available.

Applies to the `xorpd` below as well.

Reply:
We've not been doing this for helper intrinsics so far. @CarolEidt, @fiigii, does this seem like something we can/should do for the helper intrinsics specifically?

For the regular intrinsics, we have a contract mapping it to a given instruction (but not encoding). However, for the helper intrinsics, we have a contract of behavior instead, so it might make sense for these to produce the "most optimal" codegen.

Reply:
You really should print that contract on a piece of paper and burn it.